it still doesnt work
i had to repair my pc a bit now im back

```
Compiling .pioenvs/esphome-trash-can/src/esphome/components/7segment_gpio/7segment_gpio.cpp.o
Compiling .pioenvs/esphome-trash-can/src/main.cpp.o
src/esphome/components/7segment_gpio/7segment_gpio.cpp: In lambda function:
src/esphome/components/7segment_gpio/7segment_gpio.cpp:379:33: error: 'display_type' was not declared in this scope
     const auto digit_on_level = display_type == CommonAnode;
                                 ^~~~~~~~~~~~
src/esphome/components/7segment_gpio/7segment_gpio.cpp:379:33: note: suggested alternative: 'DisplayType'
     const auto digit_on_level = display_type == CommonAnode;
                                 ^~~~~~~~~~~~
                                 DisplayType
src/esphome/components/7segment_gpio/7segment_gpio.cpp: In lambda function:
src/esphome/components/7segment_gpio/7segment_gpio.cpp:384:35: error: 'display_type' was not declared in this scope
     const auto segment_on_level = display_type != CommonAnode;
                                   ^~~~~~~~~~~~
src/esphome/components/7segment_gpio/7segment_gpio.cpp:384:35: note: suggested alternative: 'DisplayType'
     const auto segment_on_level = display_type != CommonAnode;
                                   ^~~~~~~~~~~~
                                   DisplayType
src/esphome/components/7segment_gpio/7segment_gpio.cpp: In member function 'void esphome::lcd_digits::LcdDigitsComponent::set_progress(float)':
src/esphome/components/7segment_gpio/7segment_gpio.cpp:391:44: error: invalid use of void expression
       pin->digital_write(digit_level(false));
                                            ^
src/esphome/components/7segment_gpio/7segment_gpio.cpp:394:44: error: invalid use of void expression
     pin->digital_write(segment_level(false));
                                            ^
src/esphome/components/7segment_gpio/7segment_gpio.cpp:395:83: error: invalid use of void expression
   interrupt_data_.segment_pins[current_segment]->digital_write(segment_level(true));
                                                                                   ^
src/esphome/components/7segment_gpio/7segment_gpio.cpp:397:41: error: invalid use of void expression
     pin->digital_write(digit_level(true));
                                         ^
*** [.pioenvs/esphome-trash-can/src/esphome/components/7segment_gpio/7segment_gpio.cpp.o] Error 1
========================= [FAILED] Took 89.51 seconds =========================
```

--- a/components/7segment_gpio/7segment_gpio.cpp
+++ b/components/7segment_gpio/7segment_gpio.cpp
@@ -376,12 +376,12 @@ void LcdDigitsComponent::set_progress(float progress) {
   };
 
   auto digit_level = [&](bool on) {
-    const auto digit_on_level = display_type == CommonAnode;
+    const auto digit_on_level = interrupt_data_.display_type == CommonAnode;
     return invert_if_not(digit_on_level, on);
   };
 
   auto segment_level = [&](bool on) {
-    const auto segment_on_level = display_type != CommonAnode;
+    const auto segment_on_level = interrupt_data_.display_type != CommonAnode;
     return invert_if_not(segment_on_level, on);
   };
 

--- a/components/7segment_gpio/display.py
+++ b/components/7segment_gpio/display.py
@@ -15,8 +15,8 @@ DisplayType = lcd_digits_ns.enum(
     "DisplayType"
 )
 DISPLAY_TYPE = {
-    "COMMONANODE": DisplayType.CommonAnode,
-    "COMMONCATHODE": DisplayType.CommonCathode,
+    "COMMON_ANODE": DisplayType.CommonAnode,
+    "COMMON_CATHODE": DisplayType.CommonCathode,
 }
 
 # CONF_REVERSE_ENABLE = "reverse_enable"
@@ -39,7 +39,7 @@ CONFIG_SCHEMA = (
                             gpio_output_pin_schema
                         ),
             cv.Optional(CONF_INTENSITY, default=15): cv.int_range(min=0, max=15),
-            cv.Optional(CONF_DISPLAY_TYPE, default="COMMONANODE"): cv.enum(DISPLAY_TYPE, upper=True),
+            cv.Optional(CONF_DISPLAY_TYPE, default="COMMON_ANODE"): cv.enum(DISPLAY_TYPE, upper=True),
             cv.Optional(CONF_ITERATE_DIGITS, default=True): cv.boolean,
             cv.Optional(CONF_COMPENSATE_BRIGHTNESS, default=False): cv.boolean,
             cv.Optional(CONF_INTENSITY, default=15): cv.int_range(min=0, max=15),

